osu.Game/Screens/Ranking/ReplayDownloadButton.cs

```diff
@@ -99,6 +103,38 @@ private void load(OsuGame game, ScoreModelDownloader scores)
             }, true);
         }
 
+        public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
+        {
+            switch (e.Action)
+            {
+                case GlobalAction.SaveReplay:
+                    button.TriggerClick();
+                    return true;
+
+                case GlobalAction.ExportReplay:
+                    if (State.Value == DownloadState.NotDownloaded)
+                    {
+                        button.TriggerClick();
+                    }
+
+                    State.ValueChanged += importAfterDownload;
+
+                    void importAfterDownload(ValueChangedEvent<DownloadState> valueChangedEvent)
+                    {
+                        scoreManager.Export(Score.Value);
+                        State.ValueChanged -= importAfterDownload;
```

Comment:
Surely this can't work? What happens if the button starts in `NotDownloaded` state, then transitions to `Downloading`, and `ScoreManager` attempts to export a score that's not locally available yet? Same thing for `Importing`.

This should presumably have a `State.Value == DownloadState.LocallyAvailable` check.

Reply:
Yep, probably. As I mentioned, untested as of yet.

Reply:
I've refactored this to use the same logic in both places. We could probably combine the two classes to have a base class, but don't really want to attempt that in this PR.